src/test/java/org/opensearch/ad/AbstractADTest.java

```diff
@@ -223,17 +227,33 @@ protected static void tearDownThreadPool() {
         threadPool = null;
     }
 
-    public void setupTestNodes(Settings settings, TransportInterceptor transportInterceptor) {
+    /**
+     *
+     * @param transportInterceptor Interceptor to for transport requests. Used
+     *  to mock transport layer.
+     * @param nodeSettings node override of setting
+     * @param setting the supported setting set.
+     */
+    public void setupTestNodes(TransportInterceptor transportInterceptor, final Settings nodeSettings, Setting<?>... setting) {
```

Comment:
Can you explain why you've added the additional `Setting` set here to pass to `FakeNode`?

Reply:
Please see an example: https://github.com/kaituo/anomaly-detection-1/blob/multiCat4/src/test/java/org/opensearch/ad/transport/AnomalyResultTests.java#L437-L442

The added setting set is to list supported setting set.  It is used as part of testing setting changes.

Haven't sent out tests in PRs as I am still adding unit tests.

Reply:
Got it, makes sense.